src/database-assignment-schema.sql

```diff
@@ -0,0 +1,183 @@
+-- JOIN TABLES
+DROP TABLE IF EXISTS tv_compatible_ci_modules;
+DROP TABLE IF EXISTS tv_compatible_wall_brackets;
+-- SUBTYPES
+DROP TABLE IF EXISTS televisions;
+DROP TABLE IF EXISTS remote_controllers;
+DROP TABLE IF EXISTS ci_modules;
+DROP TABLE IF EXISTS wall_brackets;
+-- BASE PRODUCTS
+DROP TABLE IF EXISTS products;
+-- USER TABLES
+DROP TABLE IF EXISTS users;
+
+CREATE TABLE users
+(
+    id            SERIAL PRIMARY KEY,
+    username      VARCHAR(100) UNIQUE NOT NULL,
+    user_password VARCHAR(255)        NOT NULL, -- in realistic scenario I would store this as a secure hash, not plain text.
+    address       VARCHAR(255),
+    user_role     VARCHAR(20) CHECK (user_role IN ('admin', 'employee')),
+    pay_scale     INT                 NOT NULL CHECK (pay_scale >= 0),
+    vacation_days INT CHECK (vacation_days >= 0)
+);
+
+CREATE TABLE products
+(
+    id            SERIAL PRIMARY KEY,
+    name          VARCHAR(50) UNIQUE,
+    brand         VARCHAR(50)      NOT NULL,
+    price         DOUBLE PRECISION NOT NULL CHECK (price >= 0),
+    current_stock INT CHECK (current_stock >= 0),
+    sold          INT DEFAULT 0 CHECK (sold >= 0),
+    date_sold     DATE,
```

Comment:
Ik heb dit zelf volgens mij ook zo gedaan, maar vraag me nu tijdens het nakijken ineens af of dit datum-veld op het goede niveau staat. Want als er nog 10 van dit product in stock zijn, dan kan je er 1 verkopen. En hoe wordt dat dan geregistreerd? Of zou je dan een lijstje met datums moeten hebben?